test/fuzz/simplify.cpp

```diff
@@ -14,76 +15,73 @@ using namespace Halide::Internal;
 
 const int fuzz_var_count = 5;
 
-// use std::mt19937 instead of rand() to ensure consistent behavior on all systems
-std::mt19937 rng(0);
-
 Type fuzz_types[] = {UInt(1), UInt(8), UInt(16), UInt(32), Int(8), Int(16), Int(32)};
 const int fuzz_type_count = sizeof(fuzz_types) / sizeof(fuzz_types[0]);
 
 std::string fuzz_var(int i) {
     return std::string(1, 'a' + i);
 }
 
-Expr random_var() {
-    int fuzz_count = rng() % fuzz_var_count;
+Expr random_var(FuzzedDataProvider &fdp) {
+    int fuzz_count = fdp.ConsumeIntegralInRange<int>(0, fuzz_var_count);
     return Variable::make(Int(0), fuzz_var(fuzz_count));
 }
 
-Type random_type(int width) {
-    Type T = fuzz_types[rng() % fuzz_type_count];
+Type random_type(FuzzedDataProvider &fdp, int width) {
+    Type T = fdp.PickValueInArray(fuzz_types);
 
     if (width > 1) {
         T = T.with_lanes(width);
     }
     return T;
 }
 
-int get_random_divisor(Type t) {
+int get_random_divisor(FuzzedDataProvider &fdp, Type t) {
     std::vector<int> divisors = {t.lanes()};
     for (int dd = 2; dd < t.lanes(); dd++) {
         if (t.lanes() % dd == 0) {
             divisors.push_back(dd);
         }
     }
 
-    return divisors[rng() % divisors.size()];
+    return divisors[fdp.ConsumeIntegralInRange<size_t>(0, divisors.size() - 1)];
 }
 
-Expr random_leaf(Type T, bool overflow_undef = false, bool imm_only = false) {
+Expr random_leaf(FuzzedDataProvider &fdp, Type T, bool overflow_undef = false, bool imm_only = false) {
     if (T.is_int() && T.bits() == 32) {
         overflow_undef = true;
     }
     if (T.is_scalar()) {
-        int var = rng() % fuzz_var_count + 1;
+        int var = fdp.ConsumeIntegralInRange<int>(0, fuzz_var_count) + 1;
         if (!imm_only && var < fuzz_var_count) {
-            auto v1 = random_var();
+            auto v1 = random_var(fdp);
             return cast(T, v1);
         } else {
             if (overflow_undef) {
                 // For Int(32), we don't care about correctness during
                 // overflow, so just use numbers that are unlikely to
                 // overflow.
-                return cast(T, (int)(rng() % 256 - 128));
+                return cast(T, fdp.ConsumeIntegralInRange<int>(-128, 128));
             } else {
-                return cast(T, (int)(rng() - RAND_MAX / 2));
+                return cast(T, fdp.ConsumeIntegralInRange<int>(-RAND_MAX / 2, RAND_MAX / 2));
```

Comment:
Should be `fdp.ConsumeIntegralInRange<int>(-RAND_MAX / 2, RAND_MAX / 2 -1)` as per #7546

Reply:
LGTM

Reply:
...except, RAND_MAX applies to the `rand()` call in stdlib; does libfuzzer also make the same promise about range?